modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/upgrade/v2_0_0/SXPBlueprintUpgradeProcess.java

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ *
+ *
+ *
+ */
+
+package com.liferay.search.experiences.internal.upgrade.v2_0_0;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.StringUtil;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * @author Gustavo Lima
+ */
+public class SXPBlueprintUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		_upgradeLowLevelSearchOptionsPortlets();
+		_upgradeSearchBarPortlets();
+		_upgradeSXPBlueprintOptionsPortlets();
+	}
+
+	private String _changeJSON(String small, String erc) throws Exception {
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray(small);
+
+		JSONObject jsonObject = jsonArray.getJSONObject(0);
+
+		jsonObject.put(
+			"key", "search.experiences.blueprint.external.reference.code"
+		).put(
+			"value", erc
+		);
+
+		return StringBundler.concat(
+			StringPool.OPEN_BRACKET, jsonObject, StringPool.CLOSE_BRACKET);
+	}
+
+	private long _getSXPBlueprintIdByLargeValue(String largeValue)
+		throws Exception {
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray(
+			StringBundler.concat(
+				StringPool.OPEN_BRACKET, largeValue, StringPool.CLOSE_BRACKET));
+
+		for (int i = 0; i < jsonArray.length(); i++) {
+			JSONObject jsonObject = jsonArray.getJSONObject(i);
+
+			JSONObject attributesJSONObject = jsonObject.getJSONObject(
+				"attributes");
+
+			if ((attributesJSONObject != null) &&
+				attributesJSONObject.has("sxpBlueprintId")) {
+
+				return attributesJSONObject.getLong("sxpBlueprintId");
+			}
+		}
+
+		return 0;
+	}
+
+	private long _getSXPBlueprintIdBySmallValue(String smallValue)
+		throws Exception {
+
+		JSONArray jsonArray = JSONFactoryUtil.createJSONArray(smallValue);
+
+		JSONObject jsonObject = jsonArray.getJSONObject(0);
+
+		return jsonObject.getLong("value");
+	}
```

Comment:
@felipelorenz same here,

I see that the method _changeJSON and this one use the smallValue jsonArray, 

```java
JSONArray jsonArray = JSONFactoryUtil.createJSONArray(smallValue);

JSONObject jsonObject = jsonArray.getJSONObject(0);
```

you can reuse this for both where they are being used.